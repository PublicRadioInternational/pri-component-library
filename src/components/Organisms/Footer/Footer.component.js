/**
 * @file Footer.component.js
 * Exports the site footer.
 */

import React from 'react';
import styles from './Footer.css';

import Menu from '../../Molecules/Menu/Menu.component';

/**
 * Component that renders the footer.
 */
const Footer = () => (
  <footer className={styles.siteFooter}>
    <section className={styles.sponsors}>
      <a href="/">
        <img
          height="48"
          width="466"
          alt="Public Radio International"
          title="Public Radio International"
          typeof="foaf:Image"
          src="https://media.pri.org/s3fs-public/styles/large/public/different-voice.png"
        />
      </a>
      <p>Major funding provided by:</p>
      <a
        href="http://www.gatesfoundation.org/"
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          height="52"
          width="149"
          alt="Bill &amp;amp; Melinda Gates Foundation"
          title="Bill &amp;amp; Melinda Gates Foundation"
          typeof="foaf:Image"
          src="https://cdn1.pri.org/sites/all/themes/pri/images/logo-gates-color.jpg"
        />
      </a>
      <a
        href="https://www.macfound.org/"
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          height="52"
          width="149"
          alt="MacArthur Foundation"
          title="MacArthur Foundation"
          typeof="foaf:Image"
          src="https://media.pri.org/s3fs-public/logo-macarthur-color.jpg"
        />
      </a>
      <p>A Partner of OZY Media News</p>
    </section>

    <Menu />
  </footer>
);

export default Footer;
