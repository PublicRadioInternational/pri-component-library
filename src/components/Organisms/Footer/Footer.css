/* import colors... */
@value colors: "../../00_global/colors.css";
@value grayLight, grayLighter from colors;
/* import fonts... */
@value fonts: "../../00_global/fonts.css";
@value copy from fonts;

.siteFooter {
  background: grayLighter;
}

.sponsors {
  border-top: 1px solid #ddd;
  text-align: center;
  font: 500 0.75em copy;
  line-height: 2em;
  color: grayLight;
  padding: 1rem;
}

.sponsor {
  margin: 0 6px;
}

.footerBtm {
  border-top: 1px solid #ddd;
  color: grayLight;
  font: 500 0.75em copy;
  padding: 1rem;
  text-align: center;
}
