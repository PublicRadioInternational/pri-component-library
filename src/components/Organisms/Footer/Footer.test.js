/**
 * @file Footer.test.js
 * Contains tests for Footer.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Footer from './Footer.component';

describe('<Footer />', () => {
  it('Matches the Header snapshot', () => {
    const component = renderer.create(<Footer />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
