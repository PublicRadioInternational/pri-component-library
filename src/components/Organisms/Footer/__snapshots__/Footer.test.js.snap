// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Matches the Footer snapshot 1`] = `
<footer
  className="siteFooter"
>
  <section
    className="sponsors"
  >
    <a
      href="/"
    >
      <img
        alt="Public Radio International"
        height="48"
        src="https://media.pri.org/s3fs-public/styles/large/public/different-voice.png"
        title="Public Radio International"
        typeof="foaf:Image"
        width="466"
      />
    </a>
    <p>
      Major funding provided by:
    </p>
    <a
      href="http://www.gatesfoundation.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Bill &amp; Melinda Gates Foundation"
        height="52"
        src="https://cdn1.pri.org/sites/all/themes/pri/images/logo-gates-color.jpg"
        title="Bill &amp; Melinda Gates Foundation"
        typeof="foaf:Image"
        width="149"
      />
    </a>
    <a
      href="https://www.macfound.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="MacArthur Foundation"
        height="52"
        src="https://media.pri.org/s3fs-public/logo-macarthur-color.jpg"
        title="MacArthur Foundation"
        typeof="foaf:Image"
        width="149"
      />
    </a>
    <p>
      A Partner of OZY Media News
    </p>
  </section>
  <ul
    className="menu"
  />
</footer>
`;
